lib/marketplaceValidate.ts

```diff
@@ -75,72 +91,73 @@ const processValidationErrors = (i18nKey, validationResults) => {
+      i18n(`${i18nKey}.results.warnings.lineNumber`, {
+        line,
+      })
+    );
+  }
+}
+
+type Result = {
+  status: string;
+  results: Check[];
 };
 
-const displayValidationResults = (i18nKey, validationResults) => {
-  const displayResults = checks => {
-    const displayFileInfo = (file, line) => {
-      if (file) {
-        logger.log(
-          i18n(`${i18nKey}.results.warnings.file`, {
-            file,
-          })
-        );
-      }
-      if (line) {
-        logger.log(
-          i18n(`${i18nKey}.results.warnings.lineNumber`, {
-            line,
-          })
-        );
-      }
-      return null;
-    };
+type ValidationType = keyof GetValidationResultsResponse['results'];
 
-    if (checks) {
-      const { status, results } = checks;
+function displayResults(checks: Result, i18nKey: string): void {
+  if (checks) {
+    const { status, results } = checks;
 
-      if (status === 'FAIL') {
-        const failedValidations = results.filter(
-          test => test.status === 'FAIL'
-        );
-        const warningValidations = results.filter(
-          test => test.status === 'WARN'
-        );
-        failedValidations.forEach(val => {
-          logger.error(`${val.message}`);
-          displayFileInfo(val.file, val.line);
-        });
-        warningValidations.forEach(val => {
-          logger.warn(`${val.message}`);
-          displayFileInfo(val.file, val.line);
-        });
-      }
-      if (status === 'PASS') {
-        logger.success(i18n(`${i18nKey}.results.noErrors`));
+    if (status === 'FAIL') {
+      const failedValidations = results.filter(test => test.status === 'FAIL');
+      const warningValidations = results.filter(test => test.status === 'WARN');
+      failedValidations.forEach(val => {
+        logger.error(`${val.message}`);
+        displayFileInfo(val.file, val.line, i18nKey);
+      });
+      warningValidations.forEach(val => {
+        logger.warn(`${val.message}`);
+        displayFileInfo(val.file, val.line, i18nKey);
+      });
+    }
+    if (status === 'PASS') {
+      logger.success(i18n(`${i18nKey}.results.noErrors`));
 
-        results.forEach(test => {
-          if (test.status === 'WARN') {
-            logger.warn(`${test.message}`);
-            displayFileInfo(test.file, test.line);
-          }
-        });
-      }
+      results.forEach(test => {
+        if (test.status === 'WARN') {
+          logger.warn(`${test.message}`);
+          displayFileInfo(test.file, test.line, i18nKey);
+        }
+      });
     }
-    return null;
-  };
+  }
+  return;
+}
 
+export function displayValidationResults(
+  i18nKey: string,
+  validationResults: GetValidationResultsResponse
+) {
   Object.keys(validationResults.results).forEach(type => {
     logger.log(chalk.bold(i18n(`${i18nKey}.results.${type.toLowerCase()}`)));
-    displayResults(validationResults.results[type]);
+    displayResults(validationResults.results[type as ValidationType], i18nKey);
```

Comment:
Iterating objects in TS is super frustrating, I always think that there has to be a way to do it without asserting, and I'm always disappointed

Reply:
Yeah `Object.keys` not being able to figure out the key types is super frustrating